### PR TITLE
Move compiler to zserio-cli crate

### PR DESCRIPTION
**Pull request by wichert**
This creates a new `zserio-rs-build` crate that is explicitly for the compiler, and is self-contained in its own `crates/zserio-rs-build` folder.

To help people who have `rust_zserio` installed a transitional `rust_zserio` package is also included. Its only purpose is to tell people about the new package. As a bonus it has a dummy `rust_zserio` binary that will call `zserio-rs-build` automatically to smoothen the transition.

### CODE REVIEW

**Comment by wichert** [comment]
@Danaozhong I am not entirely happy with the name `zserio-cli`: it is not a very intuitive name for the command-line tool. Currently the naming is:

- `zserio` for the runtime code, and the crate you will import in client code to get the `from_*` and `to_*` functions to read/write zserio data
- `zserio_cli` for the compiler, and name of the command-line tool.

Alternatives could be:
- `zserio` / `zserio-rs-build`.  A longer name, but very explicitl.
- `zserio_runtime` / `zserio` . I don't really like the `_runtime` part personally.
- `zserio_support` / `zserio`

File: crates/rust-zserio/README.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# rust-zserio is obsolete
+
+This package is obsolete, and should not be used. Please use the
+[zserio-cli](https://crates.io/crates/zserio-cli) package for the Rust zserio
```
Comment (Danaozhong) [inline crates/rust-zserio/README.md:4]:
```suggestion
[zserio-cli](https://crates.io/crates/zserio-rs-build) package for the Rust zserio
```

File: crates/zserio-rs-build/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# zserio-cli
```
Comment (Danaozhong) [inline crates/zserio-rs-build/README.md:1]:
```suggestion
# zserio-rs-build
```
I think the naming is currently a inconsistent. There are some places where the crate is called `zserio-rs-build`, and in other cases `zserio-cli`.

**Review by Danaozhong — APPROVED** [review]
LGTM

File: crates/zserio-rs-build/README.md (hunk 2 above)
Comment (wichert) [reply crates/zserio-rs-build/README.md]: Good catch, there were two places that sitll used the old name. I've fixed both of them.

**Pull request merged**